src/ir/hashed.h

```diff
@@ -82,18 +82,11 @@ struct FunctionHasher : public WalkerPass<PostWalker<FunctionHasher>> {
 
   static HashType hashFunction(Function* func) {
     HashType ret = 0;
-    ret = rehash(ret, (HashType)func->getNumParams());
-    for (auto type : func->params) {
-      ret = rehash(ret, (HashType)type);
-    }
-    ret = rehash(ret, (HashType)func->getNumVars());
+    ret = rehash(ret, (HashType)func->sig.params);
+    ret = rehash(ret, (HashType)func->sig.results);
```

Comment:
In this case I believe the previous hash *was* deterministic, and now it isn't.

As before, this is probably fine, as passes should not be nondeterministic due only to hashing. But if the emscripten test suite determinism tests start to fail, this may indicate a pass does depend on hashing.

Reply:
I'll make sure to run the Emscripten test suite and the fuzzer before landing this to check for issues like that.